pkg/bugzilla/client.go

```diff
@@ -0,0 +1,80 @@
+// Package bugzilla implements a basic bugzilla jsonrpc api wrapper with a single query
+package bugzilla
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Bugzilla object with information to access the bugzilla api
+type Bugzilla struct {
+	username string
+	password string
+	url      string
+}
+
+// clientRequest provides a basic json request implementation
+type clientRequest struct {
+	Method string         `json:"method"`
+	Params [1]interface{} `json:"params"`
+	Id     uint64         `json:"id"`
+}
+
+// clientResponse provides a basic json response implementation
+type clientResponse struct {
+	Id     uint64      `json:"id"`
+	Result Bugs        `json:"result"`
+	Error  interface{} `json:"error"`
+}
+
+// NewClient creates and returns a bugzilla client
+func NewClient(user string, pass string, address string) Bugzilla {
```

Comment:
return a pointer to the client, since you're going to be calling methods on the pointer.  Better yet, define an interface, and return that, so you can easily unit test other consumers:

```go
type Client interface {
    ExecuteQuery(query string) (Bugs, error)
}

type httpBZClient struct {
  username string
  password string
  url string
}

// notice that we don't need to specify the type multiple times if adjacent parameters share a type
func NewClient(user, pass, addr string) Client {
  return &httpBZClient{
    username: user,
    password: pass,
    url: add,
  }
}
```